src/runner/basic.rs

```diff
@@ -198,18 +374,61 @@ impl<World> Default for Basic<World> {
                 .unwrap_or(ScenarioType::Concurrent)
         };
 
+        #[allow(clippy::shadow_unrelated)]
+        let retry: RetryOptionsFn = |f, r, sc, cli| {
+            let parse_tags = |tags: &[String]| {
+                tags.iter().find_map(|tag| {
+                    tag.strip_prefix("retry").map(|retries| {
+                        let (num, rest) = retries
+                            .strip_prefix('(')
+                            .and_then(|s| {
+                                s.split_once(')').and_then(|(num, rest)| {
+                                    num.parse::<usize>()
+                                        .ok()
+                                        .map(|num| (Some(num), rest))
+                                })
+                            })
+                            .unwrap_or((None, retries));
+
+                        let at =
+                            rest.strip_prefix(".after").and_then(|after| {
+                                after.strip_prefix('(').and_then(|after| {
```

Comment:
@ilslv would be nice to refactor tags parsing into a separate function on `RetryOptions`, and so cover it with unit tests.